backend/app/routes.py

```diff
@@ -26,34 +37,69 @@ def login():
     
     try:
         # Verify token
-        idinfo = id_token.verify_oauth2_token(token, requests.Request(), client_id)
+        idinfo = id_token.verify_oauth2_token(user_id_token, requests.Request(), client_id)
 
         #If token issuer not from Google, return error
         if idinfo['iss'] not in ['accounts.google.com', 'https://accounts.google.com']:
-            raise ValueError('Wrong issuer')
+            raise ValueError('Invalid token')
 
         # ID token is valid. Query database for user email and create account if necessary
         user = User.objects(email=idinfo['email']).first()
         if user is None:
-            user = User(first_name = idinfo['name'].split(' ')[0], last_name = idinfo['name'].split(' ')[1], email = idinfo['email']).save()
+            user = User(first_name = idinfo['name'].split(' ')[0], last_name = idinfo['name'].split(' ')[1], email = idinfo['email'], id_token = user_id_token, access_token = access_token).save()
+        else:
+            user.access_token = access_token    # Refresh access_token
+            user.save()
     
-        return user.toJSON()
+        return user.toJSON(), 200
 
-    except ValueError:
+    except ValueError as e:
         # Invalid token
-        return Response(301, "Error: Token is invalid").toJSON()
+        raise Error(e, status_code = 400)
 
-@routes_blueprint.route('/calendar', methods=['GET'])
-def return_calendar_data():
-        
-    # Get path to client secret file
-    if os.path.exists(os.getcwd() + '/app/client_secret.json'):
-        client_secret_path = os.getcwd() + '/app/client_secret.json'
 
-    """ EXCHANGE ACCESS_CODE FOR CREDENTIALS (TODO) """
+@routes_blueprint.route('/calendar/id=<id>&start=<start_date>&end=<end_date>', methods=['GET'])
+@cross_origin(supports_credentials=True)
+def return_calendar_data(id, start_date, end_date):
+
+    # Extract Access Token from Authorization heeader
+    user = User.objects(id_token = id).first()
+    if not user:
+        raise Error("No user with the associated id_token", status_code = 400)
+    else:
+        access_token = user.access_token
+
+    # Extract date from query string params, and switch to UTC time, and then to RFC 3339 format for Google API
+    start_date = parser.parse(start_date)
```

Comment:
these method calls should go in some kind of utility function in a different module instead of throwing them into the route function. Maybe we can use this date stuff elsewhere at some point in the app.